Sources/Fullscreen/EasterEggs/DrumMachine/DrumMachineView.swift

```diff
@@ -0,0 +1,206 @@
+        return (60.0 / Double(beatsPerMinute)) / 4
+    }
+
+    private lazy var selectorView: InstrumentSelectorView = {
+        let selectorView = InstrumentSelectorView(instrument: self.instrument)
+        selectorView.translatesAutoresizingMaskIntoConstraints = false
+        return selectorView
+    }()
+
+    private lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: self.collectionViewLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.backgroundColor = .black
+        collectionView.register(DrumPadCollectionViewCell.self)
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        return collectionView
+    }()
+
+    private lazy var collectionViewLayout: UICollectionViewLayout = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        layout.minimumInteritemSpacing = cellSpacing
+        layout.minimumLineSpacing = cellSpacing
+        layout.sectionInset = UIEdgeInsets(top: padSpacing, left: padSpacing, bottom: padSpacing, right: padSpacing)
+        return layout
+    }()
+
+    private lazy var bpmSlider: UISlider = {
+        let slider = UISlider()
+        slider.translatesAutoresizingMaskIntoConstraints = false
+        slider.minimumValue = 80
+        slider.maximumValue = 160
+        slider.tintColor = .pea
+        return slider
+    }()
+
+    // MARK: - Init
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupSubviews()
+        setupConstraints()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Actions
+
+    public func reset() {
+        stop()
+        currentPad = 0
+        compositions = makeEmptyCompositions()
+        collectionView.reloadData()
+        start()
+    }
+
+    public func start() {
+        timer = Timer.scheduledTimer(timeInterval: timeInterval, target: self, selector: #selector(update), userInfo: nil, repeats: true)
+    }
+
+    public func stop() {
+        timer?.invalidate()
+        timer = nil
+    }
+
+    // MARK: - Timer
+
+    @objc private func update() {
+        currentPad += 1
+
+        if currentPad == numberOfPads {
+            currentPad = 0
+        }
+
+        let nextCell = collectionView.cellForItem(at: IndexPath(item: currentPad, section: 0)) as? DrumPadCollectionViewCell
+
+        nextCell?.flash(withDuration: timeInterval)
+
+        for (_, instrumentComposition) in compositions.enumerated() {
+            if instrumentComposition.value[currentPad] {
+                instrumentComposition.key.play()
+            }
+        }
+    }
+
+    private func makeEmptyCompositions() -> [Instrument: [Bool]] {
+        return [
+            .kick: Array(repeating: false, count: self.numberOfPads),
```

Comment:
Is the `self.` needed here? 🤔

Reply:
It probably was a lazy var at first, but now it's not needed since it's a function.